assets/js/stripe-payment-request.js

```diff
@@ -467,6 +486,32 @@ jQuery( function( $ ) {
 			return prButton && 'function' === typeof prButton.data && prButton.data( 'isCustom' );
 		},
 
+		isBrandedPaymentRequestButton: function ( prButton ) {
+			return prButton && 'function' === typeof prButton.data && prButton.data( 'isBranded' );
+		},
+
+		shouldUseGooglePayBrand: function () {
+			return window.navigator.userAgent.match(/Chrome\/([0-9]+)\./i) && 'Google Inc.' == window.navigator.vendor;
+		},
```

Comment:
I'm wondering is it better and safer to move that detection and button rendering to the server like we do with custom button at the moment? And leverage `css_selector` param value to hook into button button rendered by the site. Personally I feel better if this kind of rendering and logic resides on the site backend.

What do you think?

Reply:
> I think rendering it in the JS is more aligned to what Stripe does when rendering their button.

Stripe is the loaded library from 3rd party resource. Don't think they have the other choice :) 
But their https request is not rendering any markup. 

Anyway, let's keep it on the client for now, but let's create an `Engineering` issue for that as well. IMO worth asking for other opinions here.

Reply:
Agreed! I've created #1212.